### PR TITLE
Physics Iteration 13

### PR DESCRIPTION
**Pull request by CedricGuillemet**
A big portion of this change has already been reviewed.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13749/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13749/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13749/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13749/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13749/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13749/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13749/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13749/merge#BCU1XR#0

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 1]
```diff
@@ -331,6 +335,14 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
         return true;
     }
 
+    /** @internal */
+    public get _rawBoundingInfo(): Nullable<BoundingInfo> {
```
Comment (carolhmj) [inline packages/dev/core/src/Meshes/abstractMesh.ts:339]: Since we're already having to add the getters and setters, we could make the property public?

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Physics/physicsHelper.ts [hunk 2]
```diff
@@ -810,6 +872,9 @@ class PhysicsVortexEvent {
      * @param force
      */
     public dispose(force: boolean = true) {
+        if (!this._cylinder) {
```
Comment (sebavan) [inline packages/dev/core/src/Physics/physicsHelper.ts:875]: I wonder if dispose should also set _cylinder to null or undefined in this case ?

File: packages/dev/core/src/Physics/v2/physicsAggregate.ts [hunk 3]
```diff
@@ -4,13 +4,15 @@ import { PhysicsShape } from "./physicsShape";
...
-import { Vector3 } from "../../Maths/math.vector";
+import { TmpVectors, Vector3 } from "../../Maths/math.vector";
 import { Scalar } from "../../Maths/math.scalar";
-import { PhysicsMotionType, ShapeType } from "./IPhysicsEnginePlugin";
+import { PhysicsMotionType, PhysicsShapeType } from "./IPhysicsEnginePlugin";
 import type { Mesh } from "../../Meshes/mesh";
 import type { Observer } from "../../Misc/observable";
 import type { Nullable } from "../../types";
 import type { Node } from "../../node";
+import type { AbstractMesh } from "core/Meshes/abstractMesh";
```
Comment (sebavan) [inline packages/dev/core/src/Physics/v2/physicsAggregate.ts:14]: Could we use relative path here ?

File: packages/dev/core/src/Physics/v2/physicsAggregate.ts [hunk 4]
```diff
@@ -4,13 +4,15 @@ import { PhysicsShape } from "./physicsShape";
...
+import { BoundingBox } from "core/Culling/boundingBox";
```
Comment (sebavan) [inline packages/dev/core/src/Physics/v2/physicsAggregate.ts:15]: Could we use relative path here ?

**Review by sebavan — CHANGES_REQUESTED** [review]
I guess the import path should be changed ?

File: packages/dev/core/src/Meshes/thinInstanceMesh.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ import { Mesh } from "../Meshes/mesh";
 import { VertexBuffer, Buffer } from "../Buffers/buffer";
 import { Matrix, Vector3, TmpVectors } from "../Maths/math.vector";
 import { Logger } from "../Misc/logger";
+import { BoundingInfo } from "core/Culling/boundingInfo";
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply packages/dev/core/src/Meshes/thinInstanceMesh.ts:6]: Fixed

File: packages/dev/core/src/Physics/physicsHelper.ts (hunk 2 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Physics/physicsHelper.ts:875]: Fixed

File: packages/dev/core/src/Meshes/abstractMesh.ts (hunk 1 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Meshes/abstractMesh.ts]: Should be fixed. Let me know if I missed your point.

**Pull request merged**